### PR TITLE
Feature: Added support for jsonnet

### PR DESCRIPTION
**Pull request by vikigenius**
This PR adds support for jsonnet using [jsonnet-mode](https://github.com/mgyucht/jsonnet-mode).

### CODE REVIEW

File: modules/lang/data/config.el [hunk 1]
```diff
@@ -37,6 +37,10 @@
   :config
   (set-electric! 'json-mode :chars '(?\n ?: ?{ ?})))
 
+(use-package! jsonnet-mode
+  :mode "\\.jsonnet\\'"
```
Comment (hlissner) [inline modules/lang/data/config.el]: The jsonnet-mode package [already creates an auto-mode-alist entry for itself](https://github.com/mgyucht/jsonnet-mode/blob/2b90b4e12a11c42df0f1e5db327a50555b6ff023/jsonnet-mode.el#L243), so `:mode` can be removed (and replaced with `:defer t`).

File: modules/lang/data/config.el [hunk 2]
```diff
@@ -37,6 +37,10 @@
...
+  :config
+  (set-electric! 'json-mode :chars '(?\n ?: ?{ ?})))
```
Comment (hlissner) [inline modules/lang/data/config.el]: Shouldn't this be:

```diff
-  (set-electric! 'json-mode :chars '(?\n ?: ?{ ?})))
+  (set-electric! 'jsonnet-mode :chars '(?\n ?: ?{ ?})))

**Comment by vikigenius** [comment]
Yeah, sorry about the 2nd one, silly of me :) . I also made the change from mode to defer and it works fine. Can you review now, it should be fine I think.

File: modules/lang/data/config.el [hunk 3]
```diff
@@ -37,6 +37,10 @@
   :config
   (set-electric! 'json-mode :chars '(?\n ?: ?{ ?})))
 
+(use-package! jsonnet-mode
+  :defer "\\.jsonnet\\'"
```
Comment (hlissner) [inline modules/lang/data/config.el]: This should be `:defer t`. I don't think `:defer` will understand a string argument.

**Comment by vikigenius** [comment]
yep, wasn't really that familiar with use-package mechanics, been using doom as just an end user so til. good to know, should be fixed now

**Comment by hlissner** [comment]
No sweat! There's no better learning experience than to simply use Emacs. Thanks for the help, it's much appreciated!

**Pull request merged**